src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -113,7 +113,10 @@ internal ZipArchiveEntry(ZipArchive archive, string entryName)
 
             _compressedSize = 0; // we don't know these yet
             _uncompressedSize = 0;
-            _externalFileAttr = 0;
+            _externalFileAttr = entryName.EndsWith(Path.DirectorySeparatorChar) || entryName.EndsWith(Path.AltDirectorySeparatorChar)
+                                ? DefaultDirectoryExternalAttributes
+                                : DefaultFileExternalAttributes;
```

Comment:
Note that this will also affect the empty directory entries created from file system:
https://github.com/dotnet/runtime/blob/fc813dd6839cf327bdd994039532bfcbbef7d4c9/src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Create.cs#L412-L414

Reply:
I remember working with zip files before and having different directory separators. I just tried a small app and it seems like no translation happens there, which is sort of expected in case you want to have the specific separators inside.

This is the output after running the app from the spoiler:
```
folder/file.txt
folder\file.txt
Done.
```

<details>
  <summary>Program.cs</summary>
  

 ```csharp
using System.IO.Compression;


await using (var fileStream = new FileStream(@"test.zip", FileMode.Create))
using (var archive = new ZipArchive(fileStream, ZipArchiveMode.Create, true))
{
    await archive.CreateSomeEntry("folder/file.txt");
    await archive.CreateSomeEntry("folder\\file.txt");
}

await using (var fileStream = new FileStream(@"test.zip", FileMode.Open))
using (var archive = new ZipArchive(fileStream, ZipArchiveMode.Read, true))
{
    foreach (var e in archive.Entries)
    {
        Console.WriteLine(e.FullName);
    }
}


Console.WriteLine("Done.");

public static class Extensions
{
    public static ZipArchiveEntry EnsureReadPermissions(this ZipArchiveEntry entry, bool folder = false)
    {
        if (!OperatingSystem.IsWindows())
        { 
            int filePermissions = 420 << 16;
            int folderPermissions = 493 << 16;
            
            entry.ExternalAttributes = folder ? folderPermissions : filePermissions;
        }

        return entry;
    }

    public static async Task CreateSomeEntry(this ZipArchive archive, string path)
    {
        ZipArchiveEntry demoFile = archive.CreateEntry(path).EnsureReadPermissions();

        await using var entryStream = demoFile.Open();
        await using var streamWriter = new StreamWriter(entryStream);
        await streamWriter.WriteAsync("Hello zip!");
    }
}
  ```
  
</details>